serverless/cmd/approve/main.go

```diff
@@ -0,0 +1,57 @@
+// Program for approving DSM crypto requests.
+
+package main
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"net/http"
+	"os"
+
+	"github.com/fortanix/sdkms-client-go/sdkms"
+	"github.com/golang/glog"
+)
+
+var (
+	dsmApiKey   = flag.String("dsm_api_key", "", "API key for accessing smartkey")
+	dsmEndpoint = flag.String("dsm_endpoint", "https://www.smartkey.io", "DSM instance to use when signing with DSM")
+	state       = flag.String("state", "", "State file for the request to approve")
+)
+
+type QuorumSigningState struct {
+	Msg []byte
+	RequestID sdkms.UUID
+}
+
+func main() {
+	flag.Parse()
+
+	data, err := os.ReadFile(*state)
+	if err != nil {
+		glog.Exitf("Unable to read state file: %q", err)
+	}
+	var request QuorumSigningState
+	err = json.Unmarshal(data, &request)
+	if err != nil {
```

Comment:
Reminder:
This does not report an error if values in the source do not correspond to values in the target.
If this is important, you can implement a function like this:
func StrictUnmarshal(data []byte, v interface{}) error {
	dec := json.NewDecoder(bytes.NewReader(data))
	dec.DisallowUnknownFields()
	return dec.Decode(v)
}

Reply:
I think it's okay not to be strict here. We actually only need the request ID in order to perform the approval in DSM, and it's okay if the integrate tool adds new fields for itself later.